lib/services/database.go

```diff
@@ -142,14 +142,73 @@ func NewDatabaseFromRDSCluster(cluster *rds.DBCluster) (types.Database, error) {
 	return types.NewDatabaseV3(types.Metadata{
 		Name:        aws.StringValue(cluster.DBClusterIdentifier),
 		Description: fmt.Sprintf("Aurora cluster in %v", metadata.Region),
-		Labels:      labelsFromRDSCluster(cluster, metadata),
+		Labels:      labelsFromRDSCluster(cluster, metadata, RDSEndpointTypePrimary),
 	}, types.DatabaseSpecV3{
 		Protocol: engineToProtocol(aws.StringValue(cluster.Engine)),
 		URI:      fmt.Sprintf("%v:%v", aws.StringValue(cluster.Endpoint), aws.Int64Value(cluster.Port)),
 		AWS:      *metadata,
 	})
 }
 
+// NewDatabaseFromRDSClusterReaderEndpoint creates a database resource from an RDS cluster reader endpoint (Aurora).
+func NewDatabaseFromRDSClusterReaderEndpoint(cluster *rds.DBCluster) (types.Database, error) {
+	metadata, err := MetadataFromRDSCluster(cluster)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return types.NewDatabaseV3(types.Metadata{
+		Name:        fmt.Sprintf("%v-%v", aws.StringValue(cluster.DBClusterIdentifier), string(RDSEndpointTypeReader)),
+		Description: fmt.Sprintf("Aurora cluster in %v (%v endpoint)", metadata.Region, string(RDSEndpointTypeReader)),
+		Labels:      labelsFromRDSCluster(cluster, metadata, RDSEndpointTypeReader),
+	}, types.DatabaseSpecV3{
+		Protocol: engineToProtocol(aws.StringValue(cluster.Engine)),
+		URI:      fmt.Sprintf("%v:%v", aws.StringValue(cluster.ReaderEndpoint), aws.Int64Value(cluster.Port)),
+		AWS:      *metadata,
+	})
+}
+
+// NewDatabasesFromRDSClusterCustomEndpoints creates database resources from RDS cluster custom endpoints (Aurora).
+func NewDatabasesFromRDSClusterCustomEndpoints(cluster *rds.DBCluster) (types.Databases, error) {
+	metadata, err := MetadataFromRDSCluster(cluster)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var errors []error
+	var databases types.Databases
+	for _, endpoint := range cluster.CustomEndpoints {
+		endpointName, err := parseRDSCustomEndpoint(aws.StringValue(endpoint))
+		if err != nil {
+			errors = append(errors, trace.Wrap(err))
+			continue
+		}
+
+		database, err := types.NewDatabaseV3(types.Metadata{
+			Name:        fmt.Sprintf("%v-%v-%v", aws.StringValue(cluster.DBClusterIdentifier), string(RDSEndpointTypeCustom), endpointName),
```

Comment:
Any reason to use `%v` instead of `%s`?

Reply:
good question. no particular reason. this function is a modified version of some existing function in the file and it is using`%v`. I don't think it makes any difference in terms of outcome. But `%s` probably reads better.